labkey-client-api/src/org/labkey/remoteapi/Connection.java

```diff
@@ -462,7 +483,23 @@ public Connection addCookie(String name, String value, String domain, String pat
         cookie.setExpiryDate(expiry);
         cookie.setSecure(isSecure);
         _httpClientContext.getCookieStore().addCookie(cookie);
+
+        // Don't use session info from different server
+        if (combineHostPath(_baseURI.getHost(), _baseURI.getPath()).equals(combineHostPath(domain, path)))
```

Comment:
Not sure, but should this be startsWith() instead of equals()?

Reply:
I did some experimentation with various server configurations. Default port, custom port, blank context path, and "/labkey" context path. I also made sure it works with various valid but slightly malformed `baseUrl` values.
`combineHostPath` cleans up and double or trailing slashes.
`startsWith` would be equivalent except for a host with two LabKey installs with overlapping context paths (i.e. `localhost/labkey` and `localhost/labkey2`). Probably an exceedingly rare scenario but 🤷‍♂️